lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -1881,19 +1875,16 @@ SILInstruction *SILCombiner::visitSelectEnumInst(SelectEnumInst *SEI) {
 }
 
 SILInstruction *SILCombiner::visitTupleExtractInst(TupleExtractInst *TEI) {
-  if (TEI->getFunction()->hasOwnership())
-    return nullptr;
-
   // tuple_extract(apply([add|sub|...]overflow(x, 0)), 1) -> 0
   // if it can be proven that no overflow can happen.
-  if (TEI->getFieldIndex() != 1)
-    return nullptr;
+  if (TEI->getFieldIndex() == 1) {
```

Comment:
I changed this optimization to be enclosed inside this single scope instead of having its properties be viral at the function level scope.